### DISCUSSION TITLE
Parameter constraints

### DISCUSSION THREAD

**Discussion by nikitrian** [meta-pytorch/botorch#1823] [Q&A] [answered]
Hello,

I am adding parameter constraints to my problem as ` inequality_constraints` and I am using `ConstrainedExpectedImprovement` as an acquisition function, in order to do that. I was wondering if I can use ` inequality_constraints` with other acquisition functions such as UCB, MaxValueEntropy and Probability of Improvement.

Thank you!

**Comment by saitcakmak** [comment]
Hi @nikitrian. The constraint in ConstrainedEI refers to the outcome constraints. Parameter constraints only affect how you optimize the acquisition function (i.e., how `optimize_acqf` functions), so the choice of the acquisition function should not matter.

**Reply by nikitrian** [reply]
I see.. but when I do that the parameter constraints are violated 
` inequality_constraints = ["(x[..., 1]-x[..., 0])"]`

`acquisition = UpperConfidenceBound(gp, inequality_constraints=inequality_constraints, beta=15, maximize=False)`

I want x0<=x1

**Reply by saitcakmak** [reply, marked as answer]
Parameter constraints should not be specified while constructing the acquisition function. You need to pass them to `optimize_acqf` using the syntax described in https://github.com/pytorch/botorch/blob/main/botorch/optim/optimize.py#L451-L456

If your goal is to optimize a given objective with several different acquisition functions, I'd highly recommend using Ax, which has a very convenient API for specifying constraints: https://ax.dev/tutorials/gpei_hartmann_service.html

We also have a tutorial showing how to use a custom BoTorch acquisition function with Ax: https://botorch.org/tutorials/custom_acquisition

**Reply by nikitrian** [reply]
Very helpful, thank you!
